src/main/java/de/tum/in/www1/artemis/service/SubmissionService.java

```diff
@@ -82,4 +92,46 @@ public void hideDetails(Submission submission, User user) {
             }
         }
     }
+
+    /**
+     * Creates a new Result object, assigns it to the given submission and stores the changes to the database. Note, that this method is also called for example submissions which
+     * do not have a participation. Therefore, we check if the given submission has a participation and only then update the participation with the new result.
+     *
+     * @param submission the submission for which a new result should be created
+     * @return the newly created result
+     */
+    public Result setNewResult(Submission submission) {
+        Result result = new Result();
+        result.setSubmission(submission);
+        submission.setResult(result);
+        if (submission.getParticipation() != null) {
+            result.setParticipation(submission.getParticipation());
+        }
+        result = resultRepository.save(result);
+        submissionRepository.save(submission);
+        return result;
+    }
+
+    /**
+     * Soft lock the submission to prevent other tutors from receiving and assessing it. We set the assessor and save the result to soft lock the assessment in the client, i.e. the client will not allow
+     * tutors to assess a submission when an assessor is already assigned. If no result exists for this submission we create one first.
+     *
+     * @param submission the submission to lock
+     */
+    final Result lockSubmission(Submission submission) {
```

Comment:
Is the access modifier intentional? Rather `public...`, right?

Reply:
Yes it is intentional. I only want to use it in the classes that inherits this service

Reply:
I resolved this in a6e4435.